### PR TITLE
Add VP8/VP9 video codecs support

### PR DESCRIPTION
**Pull request by universeroc**

### CODE REVIEW

File: server/src/config.js [hunk 1]
```diff
@@ -25,7 +25,30 @@ module.exports = Object.freeze({
...
+          'x-google-start-bitrate': 1000
+        }
+      },
+      {
+        kind: 'video',
+        mimeType: 'video/H264',
+        clockRate: 90000,
+        parameters: {
+          'packetization-mode': 1,
+          'profile-level-id': '4d0032',
```
Comment (ethand91) [inline server/src/config.js:48]: Do the majority of browsers support profile level 4d0032?

File: app/src/index.js [hunk 2]
```diff
@@ -28,6 +30,38 @@ const handleSocketClose = () => {
   document.getElementById('stopRecordButton').disabled = true;
 };
 
+const getVideoCodecs = () => {
+  let params = new URLSearchParams(location.search.slice(1));
+  let videoCodec = params.get('videocodec')
+  if (!videoCodec) {
```
Comment (ethand91) [inline app/src/index.js:36]: What is the reason that H264 is the default codec?

File: app/src/index.js [hunk 3]
```diff
@@ -28,6 +30,38 @@ const handleSocketClose = () => {
...
+              'profile-level-id': '4d0032',
+              'level-asymmetry-allowed': 1,
+              'x-google-start-bitrate': 1000
+            }
+          };
+  }
+  const codec = mediasoupConfig.router.mediaCodecs.find(c=>{
+     return ~c.mimeType.toLowerCase().indexOf(videoCodec.toLowerCase())
+  });
+  return codec !== undefined ? codec : {
```
Comment (ethand91) [inline app/src/index.js:52]: If codec is not defined, your returning the same as the above (if !videoCodec), would be better to return the default codec only once

File: server/src/config.js [hunk 4]
```diff
@@ -39,6 +62,6 @@ module.exports = Object.freeze({
   plainRtpTransport: {
     listenIp: '127.0.0.1',
     rtcpMux: true,
-    comedia: false
```
Comment (ethand91) [inline server/src/config.js:42]: unneeded space.

File: server/src/mediasoup.js [hunk 5]
```diff
@@ -40,7 +42,7 @@ module.exports.createTransport = async (transportType, router, options) => {
 
   switch (transportType) {
     case 'webRtc':
-      return await router.createWebRtcTransport(config.webRtcTransport);
```
Comment (ethand91) [inline server/src/mediasoup.js:43]: unneeded space.

File: server/src/server.js [hunk 6]
```diff
@@ -70,7 +72,7 @@ wss.on('connection', async (socket, request) => {
 
   socket.once('close', () => {
     console.log('socket::close [sessionId:%s]', socket.sessionId);
-
```
Comment (ethand91) [inline server/src/server.js:73]: unneeded space.

File: server/src/server.js [hunk 7]
```diff
@@ -233,7 +235,7 @@ const publishProducerRtpStream = async (peer, producer, ffmpegRtpCapabilities) =
   const codecs = [];
   // Codec passed to the RTP Consumer must match the codec in the Mediasoup router rtpCapabilities
   const routerCodec = router.rtpCapabilities.codecs.find(
-    codec => codec.kind === producer.kind
```
Comment (ethand91) [inline server/src/server.js:236]: unneeded space.

File: server/src/server.js [hunk 8]
```diff
@@ -247,7 +249,7 @@ const publishProducerRtpStream = async (peer, producer, ffmpegRtpCapabilities) =
   const rtpConsumer = await rtpTransport.consume({
     producerId: producer.id,
     rtpCapabilities,
-    paused: true
```
Comment (ethand91) [inline server/src/server.js:250]: unneeded space.

File: server/src/server.js [hunk 9]
```diff
@@ -287,7 +289,7 @@ const getProcess = (recordInfo) => {
     case 'GStreamer':
       return new GStreamer(recordInfo);
     case 'FFmpeg':
-    default:
```
Comment (ethand91) [inline server/src/server.js:290]: unneeded space.

File: server/src/config.js (hunk 1 above)
Comment (universeroc) [reply server/src/config.js:48]: These settings are copied from mediasoup's configure

File: app/src/index.js (hunk 2 above)
Comment (universeroc) [reply app/src/index.js:36]: It's no enough reason to choose H264 the default but there should be ONE when the parameter is not given, maybe others such vp8/vp9

File: app/src/index.js (hunk 3 above)
Comment (universeroc) [reply app/src/index.js:52]: Yeah, it's two condition:

1. no query to set the `videocodec`
2. set the videocodec but invalid

Yes these lines look not good

File: server/src/server.js (hunk 9 above)
Comment (universeroc) [reply server/src/server.js:290]: Origional code has extra space at the end of the line, my editor auto trim them

File: server/src/server.js (hunk 6 above)
Comment (universeroc) [reply server/src/server.js:73]: Origional code has extra space at the end of the line, my editor auto trim them

**Review by ethand91 — APPROVED** [review]

**Pull request merged**